vllm_spyre/v1/worker/spyre_input_batch.py

```diff
@@ -0,0 +1,467 @@
+# SPDX-License-Identifier: Apache-2.0
+# Datastructures defining an input batch
+
+# Based on vllm/vllm/v1/worker/gpu_input_batch.py
+
+from dataclasses import dataclass
+from typing import Optional, cast
+
+import numpy as np
+import torch
+from vllm.sampling_params import SamplingParams, SamplingType
+from vllm.v1.sample.metadata import SamplingMetadata
+from vllm.v1.worker.block_table import BlockTable
+
+_SAMPLING_EPS = 1e-5
+
+
+@dataclass
+class CachedRequestState:
+
+    req_id: str
+    prompt_token_ids: list[int]
+    prompt: Optional[str]
+    sampling_params: SamplingParams
+    generator: Optional[torch.Generator]
+
+    block_ids: list[int]
```

Comment:
We're not doing any kv cache management right now, so I don't think we should pull in anything related to block tables here. We can always copy the code back in if we end up needing it, but I don't know if the continuous batching and kv caching logic that we will need will be identical to the gpu code or not